src/profile/webid.test.ts

```diff
@@ -146,22 +157,62 @@ describe("getProfileAll", () => {
     };
 
     mockedFetcher.fetch.mockResolvedValueOnce(
-      new Response(undefined, { headers: { "Content-Type": "text/turtle" } })
+      new Response(await triplesToTurtle(toRdfJsQuads(MOCK_PROFILE)), {
+        headers: {
+          "Content-Type": "text/turtle",
+        },
+        url: "https://some.profile",
+      } as ResponseInit)
     );
+
+    const profileContent = buildThing({ url: "https://some.profile" })
+      .addIri(foaf.primaryTopic, MOCK_WEBID)
+      .build();
+
+    const webIdProfile = setThing(
+      mockSolidDatasetFrom(MOCK_WEBID),
+      profileContent
+    );
+    // Mock the webid unauthenticated fetch
+    const mockedUnauthFetch = jest.requireMock("cross-fetch") as {
+      fetch: jest.Mock<
+        ReturnType<typeof window.fetch>,
+        [RequestInfo, RequestInit?]
+      >;
+    };
+    mockedUnauthFetch.fetch.mockResolvedValueOnce(
+      new Response(await triplesToTurtle(toRdfJsQuads(webIdProfile)), {
+        headers: {
+          "Content-Type": "text/turtle",
+        },
+        url: MOCK_WEBID,
+      } as ResponseInit)
+    );
+
     await getProfileAll(MOCK_WEBID);
+    // The embedded fetch should have been used.
     expect(mockedFetcher.fetch).toHaveBeenCalled();
   });
 
-  it("uses the provided fetch if any", async () => {
-    const mockedFetch = jest.fn(fetch).mockResolvedValueOnce(
+  it("does not use the provided fetch to get the WebID", async () => {
+    const mockedFetcher = jest.requireMock("cross-fetch") as {
+      fetch: jest.Mock<
+        ReturnType<typeof window.fetch>,
+        [RequestInfo, RequestInit?]
+      >;
+    };
+    mockedFetcher.fetch.mockResolvedValueOnce(
       new Response(await triplesToTurtle(toRdfJsQuads(MOCK_PROFILE)), {
         headers: {
           "Content-Type": "text/turtle",
         },
-      })
+        url: "https://some.profile",
+      } as ResponseInit)
     );
+    const mockedFetch = jest.fn(fetch);
     await getProfileAll(MOCK_WEBID, { fetch: mockedFetch });
-    expect(mockedFetch).toHaveBeenCalled();
+    expect(mockedFetch).not.toHaveBeenCalled();
+    expect(mockedFetcher.fetch).toHaveBeenCalled();
```

Comment:
Ditto here, `mockedFetch` should've been called once, the built-in fetch should've been called once too, and we can assert the URLs that are being fetched align with our expectations.